l10n_co_geo/models/res_company.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+###############################################################################
+#    License, author and contributors information in:                         #
+#    __openerp__.py file at the root folder of this module.                   #
+###############################################################################
+
+from openerp import models, fields, api
+
+
+class ResCompany(models.Model):
+    _name = 'res.company'
+    _inherit = 'res.company'
+
+    country_id = fields.Many2one(
+        comodel_name='res.country',
+        related='city_id.state_id.country_id',
+        readonly=True,
+        required=False,
+        index=False,
+        default=None,
+        help=False,
+        domain=[],
+        context={},
+        ondelete='cascade',
+        auto_join=False
+    )
+
+    city = fields.Char(
+        invisible=True
+    )
+
+    state_id = fields.Many2one(
+        'res.country.state',
+        related='city_id.state_id',
+        readonly=True,
+    )
+
+    city_id = fields.Many2one(
```

Comment:
That's a nice solution, as our private code definitely depends on this functionality, we would come into license conflict, so internally we cannot use the linked module. This published code is relicensed under a private license for internal use, but we cannot do this with the other module. I'm thinking of a technical structure we can incorporate this module here, without breaking our licensing structure. Do you have any ideas?